src/AutoRest.CSharp.V3/Output/Builders/SerializationBuilder.cs

```diff
@@ -86,24 +87,38 @@ private XmlValueSerialization BuildXmlValueSerialization(Schema schema, bool isN
                     BuilderHelpers.GetSerializationFormat(schema));
         }
 
-        private JsonSerialization BuildSerialization(Schema schema, bool isNullable)
+        private JsonSerialization BuildSerialization(Schema schema, bool isNullable, VirtualParameter[] flattenedParameters)
         {
             switch (schema)
             {
                 case ConstantSchema constantSchema:
-                    return BuildSerialization(constantSchema.ValueType, constantSchema.Value.Value == null);
+                    return BuildSerialization(constantSchema.ValueType, constantSchema.Value.Value == null, Array.Empty<VirtualParameter>());
                 case ArraySchema arraySchema:
                     return new JsonArraySerialization(
                         _typeFactory.CreateType(arraySchema, isNullable),
-                        BuildSerialization(arraySchema.ElementType, false),
+                        BuildSerialization(arraySchema.ElementType, false, Array.Empty<VirtualParameter>()),
                         _typeFactory.CreateImplementationType(arraySchema, isNullable));
                 case DictionarySchema dictionarySchema:
                     return new JsonObjectSerialization(
                         _typeFactory.CreateType(dictionarySchema, isNullable),
                         Array.Empty<JsonPropertySerialization>(),
-                        new JsonDynamicPropertiesSerialization(BuildSerialization(dictionarySchema.ElementType, false)),
+                        new JsonDynamicPropertiesSerialization(BuildSerialization(dictionarySchema.ElementType, false, Array.Empty<VirtualParameter>())),
                         _typeFactory.CreateImplementationType(dictionarySchema, isNullable)
                         );
+                case ObjectSchema objectSchema when flattenedParameters.Any():
+                    return new JsonObjectSerialization(
+                        _typeFactory.CreateType(objectSchema, isNullable),
+                        flattenedParameters.Select(fp =>
+                            new JsonPropertySerialization(
+                                fp.TargetProperty.SerializedName,
+                                fp.Schema is ConstantSchema constantSchema
+                                    ? w => w.WriteConstant(BuilderHelpers.ParseConstant(constantSchema, _typeFactory))
+                                    : (CodeWriterDelegate)(w => w.Append($"{fp.CSharpName()}")),
```

Comment:
We shouldn't be using `CSharpName` anywhere outside of model building, we should pull the name of parameters or properties.

Reply:
Fixed.